src/test/regress/expected/create_null_dist_key.out

```diff
@@ -0,0 +1,1414 @@
+     0
+(1 row)
+
+	-- Doesn't fail because the SELECT didn't perform parallel execution.
+	TRUNCATE referenced_table CASCADE;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Table "referenced_table" is modified, which might lead to data inconsistencies or distributed deadlocks via parallel accesses to hash distributed tables due to foreign keys. Any parallel modification to those hash distributed tables in the same transaction can only be executed in sequential query execution mode
+NOTICE:  truncate cascades to table "referencing_table"
+DEBUG:  truncate cascades to table "referencing_table_xxxxxxx"
+DETAIL:  from localhost:xxxxx
+COMMIT;
+BEGIN;
+	UPDATE referencing_table SET value_1 = 15;
+	-- Doesn't fail because the UPDATE didn't perform parallel execution.
+	TRUNCATE referenced_table CASCADE;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Table "referenced_table" is modified, which might lead to data inconsistencies or distributed deadlocks via parallel accesses to hash distributed tables due to foreign keys. Any parallel modification to those hash distributed tables in the same transaction can only be executed in sequential query execution mode
+NOTICE:  truncate cascades to table "referencing_table"
+DEBUG:  truncate cascades to table "referencing_table_xxxxxxx"
+DETAIL:  from localhost:xxxxx
+COMMIT;
+BEGIN;
+	SELECT COUNT(*) FROM referenced_table;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+	-- Doesn't switch to sequential execution even if the referenced_table is
+	-- a reference table and referenced by a null-shard-key distributed table.
+	ALTER TABLE referencing_table ADD COLUMN X INT;
+ROLLBACK;
+BEGIN;
+	-- Switches to sequential execution because referenced_table is a reference table
+	-- and referenced by a null-shard-key distributed table.
+	UPDATE referenced_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Table "referenced_table" is modified, which might lead to data inconsistencies or distributed deadlocks via parallel accesses to hash distributed tables due to foreign keys. Any parallel modification to those hash distributed tables in the same transaction can only be executed in sequential query execution mode
+	UPDATE referencing_table SET value_1 = 15;
+ROLLBACK;
+BEGIN;
+	UPDATE referencing_table SET value_1 = 15;
+    -- Doesn't fail because prior UPDATE didn't perform parallel execution.
+    UPDATE referenced_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Table "referenced_table" is modified, which might lead to data inconsistencies or distributed deadlocks via parallel accesses to hash distributed tables due to foreign keys. Any parallel modification to those hash distributed tables in the same transaction can only be executed in sequential query execution mode
+ROLLBACK;
```

Comment:
> Given that parallel access to null-shard-key tables is not possible(?), maybe we should never switch to sequential execution for such distributed tables?

I think that'd be nicer. Otherwise, we'd have issues similar to https://github.com/citusdata/citus/issues/6729

A user would access null-shard key table, and then switch to seq. execution unnecessarily. In this regard, I think  null-shard-key dist tables should be treated as citus local tables.

Reply:
Talked to @onderkalaci today and we decided to not enforce switching to sequential execution after accessing a null-shard-key table.